ctapipe/io/containers.py

```diff
@@ -52,14 +52,24 @@ def __str__(self):
         return return_string
 
 class MCEvent(MCShowerData):
+    """
+    Storage of MC event data
+
+    Parameters
+    ----------
+
+    tel : dict of `RawCameraData` by tel_id
+        dictionary of the data for each telescope
+
+    """
     def __init__(self, name='MCEvent'):
         super().__init__(name)
-        self.add_item('photo_electrons',dict())
+        self.add_item('tel',dict())
     def __str__(self):
         return_string = super().__str__()+"\n"
         NPix=0
-        for tel in self.photo_electrons.values():
-            for pix in tel.values():
+        for t in self.tel.values():
+            for pix in t.photo_electrons:
```

Comment:
please follow pep8 naming conventions (don't use capitals for variables), so change NPix to `n_pix` or `npix`.

Also, avoid for-loops over pixels, which you don't need since they are in numpy arrays.
you can write this better as:
``` npix = np.sum(t.photo_electrons > 0) ```

or even get rid of the first loop too like this:
``` npix = np.sum([ np.sum(t.photo_electrons > 0) for t in tel.values]) ```